File: cpp/222.cpp
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
  int countNodes(TreeNode* root) {
    int lc =0, rc = 0;
    for(auto t = root; t; t = t->left) ++lc;
    for(auto t = root; t; t = t->right) ++rc;
    if(lc == rc)
      return pow(2, lc) - 1;
    else
      return 1 + countNodes(root->left) + countNodes(root->right);
  }
};

File: python/222.py
class TreeNode(object):
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution(object):
    def countNodes(self, root):
        lc = 0
        t = root
        while t:
            lc += 1
            t = t.left
        rc = 0
        t = root
        while t:
            rc += 1
            t = t.right
        if lc == rc:
            return 2**lc - 1
        else:
            return 1 + self.countNodes(root.left) + self.countNodes(root.right)
